tests/longhaul/longhaul_util.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2016 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package longhaul
+
+import (
+	"math/rand"
+
+	"github.com/coreos/pkg/capnslog"
+	"github.com/rook/rook/tests/framework/installer"
+	"github.com/rook/rook/tests/framework/utils"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"time"
+)
+
+//utility to install and promethous and grafana to monitor rook cluster
+var (
+	logger = capnslog.NewPackageLogger("github.com/rook/rook", "longhaul")
+)
+
+type ChaosHelper struct {
+	namespace string
+	kh        *utils.K8sHelper
+}
+
+func NewChaosHelper(namespace string, kh *utils.K8sHelper) *ChaosHelper {
+	return &ChaosHelper{namespace: namespace, kh: kh}
+
+}
+
+func (h *ChaosHelper) Monkey() {
+	listOpts := metav1.ListOptions{LabelSelector: "app=rook-operator"}
+	pl, err := h.kh.Clientset.CoreV1().Pods(installer.SystemNamespace(h.namespace)).List(listOpts)
+	if err != nil {
+		logger.Infof("Cannot check status of rook-operator pod,err ->%v", err)
+		return
+	}
+	creationTimeStamp := pl.Items[0].CreationTimestamp
+	t := time.Unix(0, creationTimeStamp.UnixNano())
+	for {
+		elapsed := time.Since(t)
```

Comment:
why not put the Sleep at the beginning of the loop? Then you don't need to check at all for elapsed minutes

Reply:
thought about a few more egde cases and putting the sleep defore the loop makes sense and avoids wierd edge cases - eg users runs block load test for 30 min, and then runs object.  Monkey might kill a pod while we are creating a object store causing expected failures.